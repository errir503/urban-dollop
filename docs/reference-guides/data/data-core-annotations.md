# Annotations

Namespace: `core/annotations`.

<div class="callout callout-alert">
This package is still experimental. “Experimental” means this is an early implementation subject to drastic and breaking changes.
</div>

## Selectors

<!-- START TOKEN(Autogenerated selectors|../../../packages/annotations/src/store/selectors.js) -->

Nothing to document.

<!-- END TOKEN(Autogenerated selectors|../../../packages/annotations/src/store/selectors.js) -->

## Actions

<!-- START TOKEN(Autogenerated actions|../../../packages/annotations/src/store/actions.js) -->

Nothing to document.

<!-- END TOKEN(Autogenerated actions|../../../packages/annotations/src/store/actions.js) -->
