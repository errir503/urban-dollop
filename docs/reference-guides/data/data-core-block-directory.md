# Block directory

Namespace: `core/block-directory`.

## Selectors

<!-- START TOKEN(Autogenerated selectors|../../../packages/block-directory/src/store/selectors.js) -->

### getDownloadableBlocks

Returns the available uninstalled blocks.

_Parameters_

-   _state_ `Object`: Global application state.
-   _filterValue_ `string`: Search string.

_Returns_

-   `Array`: Downloadable blocks.

### getErrorNoticeForBlock

Returns the error notice for a given block.

_Parameters_

-   _state_ `Object`: Global application state.
-   _blockId_ `string`: The ID of the block plugin. eg: my-block

_Returns_

-   `string|boolean`: The error text, or false if no error.

### getErrorNotices

Returns all block error notices.

_Parameters_

-   _state_ `Object`: Global application state.

_Returns_

-   `Object`: Object with error notices.

### getInstalledBlockTypes

Returns the block types that have been installed on the server in this session.

_Parameters_

-   _state_ `Object`: Global application state.

_Returns_

-   `Array`: Block type items

### getNewBlockTypes

Returns block types that have been installed on the server and used in the current post.

_Parameters_

-   _state_ `Object`: Global application state.

_Returns_

-   `Array`: Block type items.

### getUnusedBlockTypes

Returns the block types that have been installed on the server but are not used in the current post.

_Parameters_

-   _state_ `Object`: Global application state.

_Returns_

-   `Array`: Block type items.

### isInstalling

Returns true if a block plugin install is in progress.

_Parameters_

-   _state_ `Object`: Global application state.
-   _blockId_ `string`: Id of the block.

_Returns_

-   `boolean`: Whether this block is currently being installed.

### isRequestingDownloadableBlocks

Returns true if application is requesting for downloadable blocks.

_Parameters_

-   _state_ `Object`: Global application state.
-   _filterValue_ `string`: Search string.

_Returns_

-   `boolean`: Whether a request is in progress for the blocks list.

<!-- END TOKEN(Autogenerated selectors|../../../packages/block-directory/src/store/selectors.js) -->

## Actions

<!-- START TOKEN(Autogenerated actions|../../../packages/block-directory/src/store/actions.js) -->

### addInstalledBlockType

Returns an action object used to add a block type to the "newly installed" tracking list.

_Parameters_

-   _item_ `Object`: The block item with the block id and name.

_Returns_

-   `Object`: Action object.

### clearErrorNotice

Sets the error notice to empty for specific block.

_Parameters_

-   _blockId_ `string`: The ID of the block plugin. eg: my-block

_Returns_

-   `Object`: Action object.

### fetchDownloadableBlocks

Returns an action object used in signalling that the downloadable blocks have been requested and are loading.

_Parameters_

-   _filterValue_ `string`: Search string.

_Returns_

-   `Object`: Action object.

### installBlockType

Action triggered to install a block plugin.

_Parameters_

-   _block_ `Object`: The block item returned by search.

_Returns_

-   `boolean`: Whether the block was successfully installed & loaded.

### receiveDownloadableBlocks

Returns an action object used in signalling that the downloadable blocks have been updated.

_Parameters_

-   _downloadableBlocks_ `Array`: Downloadable blocks.
-   _filterValue_ `string`: Search string.

_Returns_

-   `Object`: Action object.

### removeInstalledBlockType

Returns an action object used to remove a block type from the "newly installed" tracking list.

_Parameters_

-   _item_ `string`: The block item with the block id and name.

_Returns_

-   `Object`: Action object.

### setErrorNotice

Sets an error notice to be displayed to the user for a given block.

_Parameters_

-   _blockId_ `string`: The ID of the block plugin. eg: my-block
-   _message_ `string`: The message shown in the notice.
-   _isFatal_ `boolean`: Whether the user can recover from the error.

_Returns_

-   `Object`: Action object.

### setIsInstalling

Returns an action object used to indicate install in progress.

_Parameters_

-   _blockId_ `string`:
-   _isInstalling_ `boolean`:

_Returns_

-   `Object`: Action object.

### uninstallBlockType

Action triggered to uninstall a block plugin.

_Parameters_

-   _block_ `Object`: The blockType object.

<!-- END TOKEN(Autogenerated actions|../../../packages/block-directory/src/store/actions.js) -->
