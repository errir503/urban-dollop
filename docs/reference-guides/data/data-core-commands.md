# The Commands Data

Namespace: `core/commands`.

## Selectors

<!-- START TOKEN(Autogenerated selectors|../../../packages/commands/src/store/selectors.js) -->

### getCommandLoaders

Returns the registered command loaders.

_Parameters_

-   _state_ `Object`: State tree.
-   _contextual_ `boolean`: Whether to return only contextual command loaders.

_Returns_

-   `import('./actions').WPCommandLoaderConfig[]`: The list of registered command loaders.

### getCommands

Returns the registered static commands.

_Parameters_

-   _state_ `Object`: State tree.
-   _contextual_ `boolean`: Whether to return only contextual commands.

_Returns_

-   `import('./actions').WPCommandConfig[]`: The list of registered commands.

### getContext

Returns whether the active context.

_Parameters_

-   _state_ `Object`: State tree.

_Returns_

-   `string`: Context.

### isOpen

Returns whether the command palette is open.

_Parameters_

-   _state_ `Object`: State tree.

_Returns_

-   `boolean`: Returns whether the command palette is open.

<!-- END TOKEN(Autogenerated selectors|../../../packages/commands/src/store/selectors.js) -->

## Actions

<!-- START TOKEN(Autogenerated actions|../../../packages/commands/src/store/actions.js) -->

### close

Closes the command palette.

_Returns_

-   `Object`: action.

### open

Opens the command palette.

_Returns_

-   `Object`: action.

### registerCommand

Returns an action object used to register a new command.

_Parameters_

-   _config_ `WPCommandConfig`: Command config.

_Returns_

-   `Object`: action.

### registerCommandLoader

Register command loader.

_Parameters_

-   _config_ `WPCommandLoaderConfig`: Command loader config.

_Returns_

-   `Object`: action.

### unregisterCommand

Returns an action object used to unregister a command.

_Parameters_

-   _name_ `string`: Command name.

_Returns_

-   `Object`: action.

### unregisterCommandLoader

Unregister command loader hook.

_Parameters_

-   _name_ `string`: Command loader name.

_Returns_

-   `Object`: action.

<!-- END TOKEN(Autogenerated actions|../../../packages/commands/src/store/actions.js) -->
