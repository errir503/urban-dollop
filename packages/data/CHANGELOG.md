<!-- Learn how to maintain this file at https://github.com/WordPress/gutenberg/tree/HEAD/packages#maintaining-changelogs. -->

## Unreleased

## 6.8.0 (2022-05-04)

## 6.7.0 (2022-04-21)

## 6.6.0 (2022-04-08)

## 6.5.0 (2022-03-23)

## 6.4.0 (2022-03-11)

## 6.3.0 (2022-02-23)

### New Features

- Enabled thunks by default for all stores and removed the `__experimentalUseThunks` flag.
- Store the resolution errors in store metadata and expose them using `hasResolutionFailed` the `getResolutionError` meta-selectors ([#38669](https://github.com/WordPress/gutenberg/pull/38669)).
- Expose the resolution status (undefined, resolving, finished, error) via the `getResolutionState` meta-selector ([#38669](https://github.com/WordPress/gutenberg/pull/38669)).

## 6.2.1 (2022-02-10)

### Bug Fix

-   Removed unused `memize` dependency ([#38388](https://github.com/WordPress/gutenberg/pull/38388)).

## 6.2.0 (2022-01-27)

### Bug Fix

-   Corrected expect type of action creators and selectors in Redux store configuration type
-   Move `redux` to regular dependencies and update it to version `^4.1.2`.

### Internal

-   Changed names of store-related types to better reflect their use and role.
-   Changed "storeDefinition" to "storeDescriptor" to better reflect its use and role.

## 6.1.0 (2021-09-09)

### New Features

-   Added a `batch` registry method to batch dispatch calls for performance reasons.
-   Add a new migration for the persistence plugin to migrate edit-widgets preferences to the interface package. As part of this change deprecated migrations for the persistence plugin have been removed ([#33774](https://github.com/WordPress/gutenberg/pull/33774)).
-   Update data controls to accept a data store definition as their first param in addition to a string-based store name value ([#34170](https://github.com/WordPress/gutenberg/pull/34170)).

## 6.0.0 (2021-07-29)

### Breaking Change

-   Upgraded React components to work with v17.0 ([#29118](https://github.com/WordPress/gutenberg/pull/29118)). There are no new features in React v17.0 as explained in the [blog post](https://reactjs.org/blog/2020/10/20/react-v17.html).

## 5.2.0 (2021-07-21)

## 5.1.0 (2021-05-20)

## 5.0.0 (2021-05-14)

### Breaking Changes

-   Drop support for Internet Explorer 11 ([#31110](https://github.com/WordPress/gutenberg/pull/31110)). Learn more at https://make.wordpress.org/core/2021/04/22/ie-11-support-phase-out-plan/.
-   Increase the minimum Node.js version to v12 matching Long Term Support releases ([#31270](https://github.com/WordPress/gutenberg/pull/31270)). Learn more at https://nodejs.org/en/about/releases/.

### New Features

-   Added new `startResolutions` and `finishResolutions` actions as batched variants of `startResolution` and `finishResolutions` actions.

### Enhancements

-   Updated `redux` dependency from requiring `^4.0.0` to `^4.1.0` (see changes at https://github.com/reduxjs/redux/releases/tag/v4.1.0).

## 4.27.0 (2021-03-17)

### New Features

-   Added new `resolveSelect` registry method to initiate and wait for selector resolution

## 4.26.0 (2020-12-17)

### New Features

-   Added new `register` function for registering a standard `@wordpress/data` store definition ([#26655](https://github.com/WordPress/gutenberg/pull/26655)).
-   Added new `createReduxStore` factory function that creates a data store definition for the provided Redux store options to use with `register` function ([#26655](https://github.com/WordPress/gutenberg/pull/26655)).
-   Extended `select` and `dispatch` functions to accept a data store definition as their first param in addition to a string-based store name value [#26655](https://github.com/WordPress/gutenberg/pull/26655)).
-   Extended `useDispatch` hook to accept a data store definition as their first param in addition to a string-based store name value [#26655](https://github.com/WordPress/gutenberg/pull/26655)).

### Deprecations

-   `registerGenericStore` has been deprecated. Use `register` instead.
-   `registerStore` has been deprecated. Use `register` instead.

## 4.6.0 (2019-06-12)

### New Feature

-   Expose `useSelect` hook for usage in functional components. ([#15737](https://github.com/WordPress/gutenberg/pull/15737))
-   Expose `useDispatch` hook for usage in functional components. ([#15896](https://github.com/WordPress/gutenberg/pull/15896))

### Enhancements

-   `withSelect` internally uses the new `useSelect` hook. ([#15737](https://github.com/WordPress/gutenberg/pull/15737). **Note:** This _could_ impact performance of code using `withSelect` in edge-cases. To avoid impact, memoize passed in `mapSelectToProps` callbacks or implement `useSelect` directly with dependencies.
-   `withDispatch` internally uses a new `useDispatchWithMap` hook (an internal only api) ([#15896](https://github.com/WordPress/gutenberg/pull/15896))

## 4.5.0 (2019-05-21)

### Bug Fix

-   Restore functionality of action-generators returning a Promise. Clarify intent and behaviour for `dispatch` behaviour. Dispatch actions now always
    return a promise ([#14830](https://github.com/WordPress/gutenberg/pull/14830)

### Enhancements

-   Expose `hasResolver` property on returned selectors indicating whether the selector has a corresponding resolver.

## 4.3.0 (2019-03-06)

### Enhancements

-   The `registerStore` function now accepts an optional `initialState` option value.
-   Introduce new `invalidateResolutionForStore` dispatch action for signalling to invalidate the resolution cache for an entire given store.
-   Introduce new `invalidateResolutionForStoreSelector` dispatch action for signalling to invalidate the resolution cache for a store selector (and all variations of arguments on that selector).

### Bug Fix

-   Resolves issue in the persistence plugin where passing `persist` as an array of reducer keys would wrongly replace state values for the unpersisted reducer keys.
-   Restores a behavior in the persistence plugin where a default state provided as an object will be deeply merged as a base for the persisted value. This allows for a developer to include additional new keys in a persisted value default in future iterations of their store.

## 4.2.0 (2019-01-03)

### Enhancements

-   Optimized performance of selector execution (~511% improvement)

## 4.1.0 (2018-12-12)

### New Feature

-   `withDispatch`'s `mapDispatchToProps` function takes the `registry` object as the 3rd param ([#11851](https://github.com/WordPress/gutenberg/pull/11851)).
-   `withSelect`'s `mapSelectToProps` function takes the `registry` object as the 3rd param ([#11851](https://github.com/WordPress/gutenberg/pull/11851)).

## 4.0.1 (2018-11-20)

## 4.0.0 (2018-11-15)

### Breaking Changes

-   `registry.registerReducer` has been removed. Use `registry.registerStore` instead.
-   `registry.registerSelectors` has been removed. Use `registry.registerStore` instead.
-   `registry.registerActions` has been removed. Use `registry.registerStore` instead.
-   `registry.registerResolvers` has been removed. Use `registry.registerStore` instead.

### Bug Fix

-   Resolve an issue where `withSelect`'s `mapSelectToProps` would not be rerun if the wrapped component had incurred a store change during its mount lifecycle.

## 3.1.2 (2018-11-09)

## 3.1.1 (2018-11-09)

## 3.1.0 (2018-11-03)

### New Features

-   `registry.registerGenericStore` has been added to support integration with existing data systems.

### Deprecations

-   `registry.registerReducer` has been deprecated. Use `registry.registerStore` instead.
-   `registry.registerSelectors` has been deprecated. Use `registry.registerStore` instead.
-   `registry.registerActions` has been deprecated. Use `registry.registerStore` instead.
-   `registry.registerResolvers` has been deprecated. Use `registry.registerStore` instead.

## 3.0.1 (2018-10-30)

### Internal

-   Replace Redux implementation of `combineReducers` with in-place-compatible `turbo-combine-reducers`.

## 3.0.0 (2018-10-29)

### Breaking Changes

-   Writing resolvers as async generators has been removed. Use the controls plugin instead.

## 2.1.4 (2018-10-19)

## 2.1.3 (2018-10-18)

## 2.1.0 (2018-09-30)

### New Features

-   Adding support for using controls in resolvers using the controls plugin.

### Polish

-   Updated `redux` dependency to the latest version.

### Deprecations

-   Writing resolvers as async generators has been deprecated. Use the controls plugin instead.

### Bug Fixes

-   Fix the promise middleware in Firefox.

## 2.0.0 (2018-09-05)

### Breaking Change

-   The `withRehdyration` function is removed. Use the persistence plugin instead.
-   The `loadAndPersist` function is removed. Use the persistence plugin instead.
-   `restrictPersistence`, `setPersistenceStorage` and `setupPersistence` functions have been removed. Please use the data persistence plugin instead.
-   Change how required built-ins are polyfilled with Babel 7 ([#9171](https://github.com/WordPress/gutenberg/pull/9171)). If you're using an environment that has limited or no support for ES2015+ such as lower versions of IE then using [core-js](https://github.com/zloirock/core-js) or [@babel/polyfill](https://babeljs.io/docs/en/next/babel-polyfill) will add support for these methods.
