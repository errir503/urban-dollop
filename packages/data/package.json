{
	"name": "@wordpress/data",
	"version": "5.1.0",
	"description": "Data module for WordPress.",
	"author": "The WordPress Contributors",
	"license": "GPL-2.0-or-later",
	"keywords": [
		"wordpress",
		"gutenberg",
		"data",
		"redux"
	],
	"homepage": "https://github.com/WordPress/gutenberg/tree/HEAD/packages/data/README.md",
	"repository": {
		"type": "git",
		"url": "https://github.com/WordPress/gutenberg.git",
		"directory": "packages/data"
	},
	"bugs": {
		"url": "https://github.com/WordPress/gutenberg/issues"
	},
	"engines": {
		"node": ">=12"
	},
	"main": "build/index.js",
	"module": "build-module/index.js",
	"react-native": "src/index",
	"sideEffects": false,
	"dependencies": {
		"@babel/runtime": "^7.13.10",
		"@wordpress/compose": "file:../compose",
		"@wordpress/deprecated": "file:../deprecated",
		"@wordpress/element": "file:../element",
		"@wordpress/is-shallow-equal": "file:../is-shallow-equal",
		"@wordpress/priority-queue": "file:../priority-queue",
		"@wordpress/redux-routine": "file:../redux-routine",
		"equivalent-key-map": "^0.2.2",
		"is-promise": "^4.0.0",
		"lodash": "^4.17.21",
		"memize": "^1.1.0",
		"redux": "^4.1.0",
		"turbo-combine-reducers": "^1.0.2",
		"use-memo-one": "^1.1.1"
	},
	"publishConfig": {
		"access": "public"
	}
}
