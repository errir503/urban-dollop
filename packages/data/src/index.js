/**
 * External dependencies
 */
import combineReducers from 'turbo-combine-reducers';

/**
 * Internal dependencies
 */
import defaultRegistry from './default-registry';
import * as plugins from './plugins';

/** @typedef {import('./types').WPDataStore} WPDataStore */

export { default as withSelect } from './components/with-select';
export { default as withDispatch } from './components/with-dispatch';
export { default as withRegistry } from './components/with-registry';
export {
	RegistryProvider,
	RegistryConsumer,
	useRegistry,
} from './components/registry-provider';
export { default as useSelect } from './components/use-select';
export { useDispatch } from './components/use-dispatch';
export { AsyncModeProvider } from './components/async-mode-provider';
export { createRegistry } from './registry';
export { createRegistrySelector, createRegistryControl } from './factory';
export { controls } from './controls';
export { default as createReduxStore } from './redux-store';

/**
 * Object of available plugins to use with a registry.
 *
 * @see [use](#use)
 *
 * @type {Object}
 */
export { plugins };

/**
 * The combineReducers helper function turns an object whose values are different
 * reducing functions into a single reducing function you can pass to registerReducer.
 *
 * @param {Object} reducers An object whose values correspond to different reducing
 *                          functions that need to be combined into one.
 *
 * @example
 * ```js
 * import { combineReducers, createReduxStore, register } from '@wordpress/data';
 *
 * const prices = ( state = {}, action ) => {
 * 	return action.type === 'SET_PRICE' ?
 * 		{
 * 			...state,
 * 			[ action.item ]: action.price,
 * 		} :
 * 		state;
 * };
 *
 * const discountPercent = ( state = 0, action ) => {
 * 	return action.type === 'START_SALE' ?
 * 		action.discountPercent :
 * 		state;
 * };
 *
 * const store = createReduxStore( 'my-shop', {
 * 	reducer: combineReducers( {
 * 		prices,
 * 		discountPercent,
 * 	} ),
 * } );
 * register( store );
 * ```
 *
 * @return {Function} A reducer that invokes every reducer inside the reducers
 *                    object, and constructs a state object with the same shape.
 */
export { combineReducers };

/**
 * Given the name or definition of a registered store, returns an object of the store's selectors.
 * The selector functions are been pre-bound to pass the current state automatically.
 * As a consumer, you need only pass arguments of the selector, if applicable.
 *
 * @param {string|WPDataStore} storeNameOrDefinition Unique namespace identifier for the store
 *                                                   or the store definition.
 *
 * @example
 * ```js
 * import { select } from '@wordpress/data';
 *
 * select( 'my-shop' ).getPrice( 'hammer' );
 * ```
 *
 * @return {Object} Object containing the store's selectors.
 */
export const select = defaultRegistry.select;

/**
 * Given the name of a registered store, returns an object containing the store's
 * selectors pre-bound to state so that you only need to supply additional arguments,
 * and modified so that they return promises that resolve to their eventual values,
 * after any resolvers have ran.
 *
 * @param {string|WPDataStore} storeNameOrDefinition Unique namespace identifier for the store
 *                                                   or the store definition.
 *
 * @example
 * ```js
 * import { resolveSelect } from '@wordpress/data';
 *
 * resolveSelect( 'my-shop' ).getPrice( 'hammer' ).then(console.log)
 * ```
 *
 * @return {Object} Object containing the store's promise-wrapped selectors.
 */
export const resolveSelect = defaultRegistry.resolveSelect;

/**
 * Given the name of a registered store, returns an object of the store's action creators.
 * Calling an action creator will cause it to be dispatched, updating the state value accordingly.
 *
 * Note: Action creators returned by the dispatch will return a promise when
 * they are called.
 *
 * @param {string|WPDataStore} storeNameOrDefinition Unique namespace identifier for the store
 *                                                   or the store definition.
 *
 * @example
 * ```js
 * import { dispatch } from '@wordpress/data';
 *
 * dispatch( 'my-shop' ).setPrice( 'hammer', 9.75 );
 * ```
 * @return {Object} Object containing the action creators.
 */
export const dispatch = defaultRegistry.dispatch;

/**
 * Given a listener function, the function will be called any time the state value
 * of one of the registered stores has changed. This function returns a `unsubscribe`
 * function used to stop the subscription.
 *
 * @param {Function} listener Callback function.
 *
 * @example
 * ```js
 * import { subscribe } from '@wordpress/data';
 *
 * const unsubscribe = subscribe( () => {
 * 	// You could use this opportunity to test whether the derived result of a
 * 	// selector has subsequently changed as the result of a state update.
 * } );
 *
 * // Later, if necessary...
 * unsubscribe();
 * ```
 */
export const subscribe = defaultRegistry.subscribe;

/**
 * Registers a generic store.
 *
 * @deprecated Use `register` instead.
 *
 * @param {string} key    Store registry key.
 * @param {Object} config Configuration (getSelectors, getActions, subscribe).
 */
export const registerGenericStore = defaultRegistry.registerGenericStore;

/**
 * Registers a standard `@wordpress/data` store.
 *
 * @deprecated Use `register` instead.
 *
 * @param {string} storeName Unique namespace identifier for the store.
 * @param {Object} options   Store description (reducer, actions, selectors, resolvers).
 *
 * @return {Object} Registered store object.
 */
export const registerStore = defaultRegistry.registerStore;

/**
 * Extends a registry to inherit functionality provided by a given plugin. A
 * plugin is an object with properties aligning to that of a registry, merged
 * to extend the default registry behavior.
 *
 * @param {Object} plugin Plugin object.
 */
export const use = defaultRegistry.use;

/**
 * Registers a standard `@wordpress/data` store definition.
 *
 * @example
 * ```js
 * import { createReduxStore, register } from '@wordpress/data';
 *
 * const store = createReduxStore( 'demo', {
 *     reducer: ( state = 'OK' ) => state,
 *     selectors: {
 *         getValue: ( state ) => state,
 *     },
 * } );
 * register( store );
 * ```
 *
 * @param {WPDataStore} store Store definition.
 */
export const register = defaultRegistry.register;
