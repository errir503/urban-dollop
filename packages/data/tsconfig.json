{
	"extends": "../../tsconfig.base.json",
	"compilerOptions": {
		"rootDir": "src",
		"declarationDir": "build-types",
		"noUnusedParameters": false,
		"checkJs": false
	},
	"references": [
		{ "path": "../compose" },
		{ "path": "../deprecated" },
		{ "path": "../element" },
		{ "path": "../private-apis" },
		{ "path": "../is-shallow-equal" },
		{ "path": "../priority-queue" },
		{ "path": "../redux-routine" }
	],
	"include": [ "src/**/*" ]
}
