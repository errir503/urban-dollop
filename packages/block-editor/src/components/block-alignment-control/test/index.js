/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * WordPress dependencies
 */
import { useSelect } from '@wordpress/data';

/**
 * Internal dependencies
 */
import BlockAlignmentUI from '../ui';

jest.mock( '@wordpress/data/src/components/use-select', () => {
	// This allows us to tweak the returned value on each test.
	const mock = jest.fn();
	return mock;
} );
useSelect.mockImplementation( () => ( { wideControlsEnabled: false } ) );

describe( 'BlockAlignmentUI', () => {
	const alignment = 'left';
	const onChange = jest.fn();

	const wrapper = shallow(
		<BlockAlignmentUI value={ alignment } onChange={ onChange } isToolbar />
	);

	const controls = wrapper.props().controls;

	afterEach( () => {
		onChange.mockClear();
	} );

	test( 'should match snapshot', () => {
		expect( wrapper ).toMatchSnapshot();
	} );

	test( 'should call onChange with undefined, when the control is already active', () => {
		const activeControl = controls.find(
			( { title } ) => title === 'Align left'
		);
		activeControl.onClick();

		expect( activeControl.isActive ).toBe( true );
		expect( onChange ).toHaveBeenCalledTimes( 1 );
		expect( onChange ).toHaveBeenCalledWith( undefined );
	} );

	test( 'should call onChange with alignment value when the control is inactive', () => {
		const inactiveCenterControl = controls.find(
			( { title } ) => title === 'Align center'
		);
		inactiveCenterControl.onClick();

		expect( inactiveCenterControl.isActive ).toBe( false );
		expect( onChange ).toHaveBeenCalledTimes( 1 );
		expect( onChange ).toHaveBeenCalledWith( 'center' );
	} );
} );
