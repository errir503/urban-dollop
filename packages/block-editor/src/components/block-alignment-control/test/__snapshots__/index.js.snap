// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockAlignmentUI should match snapshot 1`] = `
<ToolbarGroup
  controls={
    Array [
      Object {
        "icon": <SVG
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <Path
            d="M5 15h14V9H5v6zm0 4.8h14v-1.5H5v1.5zM5 4.2v1.5h14V4.2H5z"
          />
        </SVG>,
        "isActive": false,
        "onClick": [Function],
        "role": "menuitemradio",
        "title": "None",
      },
      Object {
        "icon": <SVG
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <Path
            d="M4 9v6h14V9H4zm8-4.8H4v1.5h8V4.2zM4 19.8h8v-1.5H4v1.5z"
          />
        </SVG>,
        "isActive": true,
        "onClick": [Function],
        "role": "menuitemradio",
        "title": "Align left",
      },
      Object {
        "icon": <SVG
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <Path
            d="M7 9v6h10V9H7zM5 19.8h14v-1.5H5v1.5zM5 4.3v1.5h14V4.3H5z"
          />
        </SVG>,
        "isActive": false,
        "onClick": [Function],
        "role": "menuitemradio",
        "title": "Align center",
      },
      Object {
        "icon": <SVG
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <Path
            d="M6 15h14V9H6v6zm6-10.8v1.5h8V4.2h-8zm0 15.6h8v-1.5h-8v1.5z"
          />
        </SVG>,
        "isActive": false,
        "onClick": [Function],
        "role": "menuitemradio",
        "title": "Align right",
      },
    ]
  }
  icon={
    <SVG
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <Path
        d="M4 9v6h14V9H4zm8-4.8H4v1.5h8V4.2zM4 19.8h8v-1.5H4v1.5z"
      />
    </SVG>
  }
  isCollapsed={true}
  label="Align"
  popoverProps={
    Object {
      "isAlternate": true,
    }
  }
  toggleProps={
    Object {
      "describedBy": "Change alignment",
    }
  }
/>
`;
